rtrlib/rtr/rtr.c

```diff
@@ -52,6 +54,99 @@ static int install_sig_handler()
     return sigaction(SIGUSR1, &sa, NULL);
 }
 
+int rtr_check_interval_range(uint32_t interval, enum interval_type type)
+{
+    switch(type) {
+    case EXPIRATION:
+        if(interval < RTR_EXPIRATION_MIN) {
+            return BELOW_INTERVAL_RANGE;
+        } else if (interval > RTR_EXPIRATION_MAX) {
+            return ABOVE_INTERVAL_RANGE;
+        }
+        break;
+    case REFRESH:
+        if(interval < RTR_REFRESH_MIN) {
+            return BELOW_INTERVAL_RANGE;
+        } else if (interval > RTR_REFRESH_MAX) {
+            return ABOVE_INTERVAL_RANGE;
+        }
+        break;
+    case RETRY:
+        if(interval < RTR_RETRY_MIN) {
+            return BELOW_INTERVAL_RANGE;
+        } else if(interval > RTR_RETRY_MAX) {
+            return ABOVE_INTERVAL_RANGE;
+        }
+        break;
+    default:
+        break;
+    }
+
+    return INSIDE_INTERVAL_RANGE;
+}
+
+void apply_interval_value(struct rtr_socket *rtr_socket,
+                          uint32_t interval,
+                          enum interval_type type)
+{
+    if (type == EXPIRATION) {
+        rtr_socket->expire_interval = interval;
+    } else if (type == REFRESH) {
+        rtr_socket->refresh_interval = interval;
+    } else if (type == RETRY) {
+        rtr_socket->retry_interval = interval;
+    }
+}
+
+int rtr_check_interval_option(struct rtr_socket *rtr_socket,
+                                     int interval_mode,
+                                     uint32_t interval,
+                                     enum interval_type type) {
+    uint16_t minimum;
+    uint32_t maximum;
+
+    switch (type) {
+    case EXPIRATION:
+        minimum = RTR_EXPIRATION_MIN;
+        maximum = RTR_EXPIRATION_MAX;
+        break;
+    case REFRESH:
+        minimum = RTR_REFRESH_MIN;
+        maximum = RTR_REFRESH_MAX;
+        break;
+    case RETRY:
+        minimum = RTR_RETRY_MIN;
+        maximum = RTR_RETRY_MAX;
+        break;
+    default:
+        break;
+    }
+
+    int interv_retval = rtr_check_interval_range(interval, type);
+
+    if (interv_retval == INSIDE_INTERVAL_RANGE || interval_mode == ACCEPT_ANY) {
```

Comment:
I would move the `switch() case` here, and check for `INSIDE_INTERVAL_RANGE` where required